Source/Core/DiscIO/SplitFileBlob.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Common/IOFile.h"
+#include "DiscIO/Blob.h"
+
+namespace DiscIO
+{
+class SplitPlainFileReader final : public BlobReader
+{
+public:
+  static std::unique_ptr<SplitPlainFileReader> Create(const std::string& first_file_path);
+
+  BlobType GetBlobType() const override { return BlobType::SPLIT_PLAIN; }
+
+  u64 GetRawSize() const override { return m_size; }
+  u64 GetDataSize() const override { return m_size; }
+  DataSizeType GetDataSizeType() const override { return DataSizeType::Accurate; }
+
+  u64 GetBlockSize() const override { return 0; }
+  bool HasFastRandomAccessInBlock() const override { return true; }
+  std::string GetCompressionMethod() const override { return {}; }
+  std::optional<int> GetCompressionLevel() const override { return std::nullopt; }
+
+  bool Read(u64 offset, u64 nbytes, u8* out_ptr) override;
+
+private:
+  struct SingleFile
```

Comment:
Any benefit in making this `final`?

Reply:
Are we worried about in being derived from?  Not aware of any benefit otherwise?

Reply:
No, mainly making sure it doesn't do anything stupid during use (and maybe even get a hint of optimization somewhere). Plus, I don't think we _intend_ for it to be derived from, so...mainly throwing things at the wall to see what sticks, I don't actually _know_ if it makes a difference.

Reply:
> Not aware of any benefit otherwise

Shouldn't compilers take advantage of `final` to optimise away the vtable?

Reply:
Yes.  But AFAIK that only applies when you have a pointer to a derived type that is referenced by the base type.  This struct isn't even polymorphic.

Reply:
Yes, `final` can be used by the compiler to optimize virtual function calls... but if there's no virtual functions in the first place there's no difference.